scripts/fiximports/main.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"golang.org/x/tools/imports"
+)
+
+var (
+	// groupByPrefixes is the list of import prefixes that should _each_ be grouped separately.
+	// See: imports.LocalPrefix.
+	groupByPrefixes = []string{
+		"github.com/filecoin-project",
+		"github.com/filecoin-project/lotus",
+	}
+	newline                  = []byte("\n")
+	importBlockRegex         = regexp.MustCompile(`(?s)import\s*\((.*?)\)`)
+	consecutiveNewlinesRegex = regexp.MustCompile(`\n\s*\n`)
+)
+
+func main() {
+	if err := filepath.Walk(".", func(path string, info fs.FileInfo, err error) error {
+		switch {
+		case err != nil:
+			return err
+		case // Skip ffi and test vectors in "./extern" directory and their contents.
+			strings.HasSuffix(path, "extern/filecoin-ffi"),
+			strings.HasSuffix(path, "extern/test-vectors"):
+			return filepath.SkipDir
+		case // Skip directories, generated cborgen go files and any other non-go files.
+			info.IsDir(),
+			strings.HasSuffix(info.Name(), "_cbor_gen.go"),
+			!strings.HasSuffix(info.Name(), ".go"):
+			return nil
+		}
+		return fixGoImports(path)
+	}); err != nil {
+		fmt.Printf("Error fixing go imports: %v\n", err)
+	}
+}
+
+func fixGoImports(path string) error {
+	sourceFile, err := os.Open(path)
+	if err != nil {
+		return err
+	}
+	defer func() { _ = sourceFile.Close() }()
+
+	source, err := io.ReadAll(sourceFile)
+	if err != nil {
+		return err
+	}
+	formatted := collapseImportNewlines(source)
+	for _, prefix := range groupByPrefixes {
+		imports.LocalPrefix = prefix
+		formatted, err = imports.Process(path, formatted, nil)
+		if err != nil {
+			return err
+		}
+	}
+	if !bytes.Equal(source, formatted) {
+		_ = sourceFile.Close()
+		if err := replaceFileContent(path, formatted); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func replaceFileContent(path string, replacement []byte) error {
+	target, err := os.OpenFile(path, os.O_RDWR|os.O_TRUNC, 0666)
```

Comment:
Why re-open? We can open read/write then call `file.Seek(0, 0)` and `file.Truncate(0)`.

Reply:
Just to avoid opening files in RW when they might not need to be. In terms of speed it made little difference on my machine. 

Happy to change it if you think the implementation is too defensive.

Reply:
Honestly, opening RW is safer IMO (e.g., if the user calls `git checkout` while running make, we'll clobber the now deleted file instead of clobbering the newly checked out file).